jwst/regtest/test_miri_mrs.py

```diff
@@ -39,5 +107,49 @@ def run_spec2(jail, rtdata_module):
 )
 def test_spec2(run_spec2, fitsdiff_default_kwargs, suffix):
     """Test ensuring the callwebb_spec2 is operating appropriately for MIRI MRS data"""
-    rt.is_like_truth(run_spec2, fitsdiff_default_kwargs, suffix,
-                     truth_path='truth/test_miri_mrs')
+    rtdata, asn_path = run_spec2
+    rt.is_like_truth(rtdata, fitsdiff_default_kwargs, suffix,
+                     truth_path=TRUTH_PATH)
+
+
+@pytest.mark.bigdata
+@pytest.mark.parametrize(
+    'output',
+    [
+        'ifushort_ch12_spec3_ch1-medium_s3d.fits',
+        'ifushort_ch12_spec3_ch1-medium_x1d.fits',
+        'ifushort_ch12_spec3_ch2-medium_s3d.fits',
+        'ifushort_ch12_spec3_ch2-medium_x1d.fits',
+        'ifushort_ch12_spec3_mrs_imatch.fits'
+    ]
+)
+def test_spec3(run_spec3, fitsdiff_default_kwargs, output):
+    """Regression test matching output files"""
+    rt.is_like_truth(
+        run_spec3, fitsdiff_default_kwargs, output,
+        truth_path=TRUTH_PATH,
+        is_suffix=False
+    )
+
+
+@pytest.mark.bigdata
+@pytest.mark.parametrize(
+    'output',
+    [
+        'ifushort_set2_0_a3001_crf.fits',
+        'ifushort_set2_0_mrs_imatch.fits',
+        'ifushort_set2_1_a3001_crf.fits',
+        'ifushort_set2_1_mrs_imatch.fits',
+        'ifushort_set2_ch1-short_s3d.fits',
+        'ifushort_set2_ch1-short_x1d.fits',
+        'ifushort_set2_ch2-short_s3d.fits',
+        'ifushort_set2_ch2-short_x1d.fits',
+    ]
+)
+def test_spec3_multi(run_spec3_multi, fitsdiff_default_kwargs, output):
+    """Regression test matching output files"""
+    rt.is_like_truth(
```

Comment:
What does the `rt.is_like_truth` traceback look like when it fails?

Is there a good reason to have it in the `RegtestData` class as opposed to separate helper, i.e. a fixture or just an assertion helper?

Reply:
First note that this is not in `RegtestData` class, but in module `regtestdata.py`. As such, it is a helper to and a user of the `RegtestData` class, and hence why it is in the module. However, no strong feeling as to where.

Reply:
Doh!  My bad.  Carry on.  👍

Reply:
How long does this test take ? I may be a good idea to test an IFULONG as well. The IFULONG is a different detector. The reference files for the IFULONG is a different set of columns. It would be nice to have an end-to-end (spec2 and spec3) test of the IFULONG.

Reply:
Not sure if end-to-end of IFULONG is necessary. However, would it affect how the "full spec3" test? I believe that would be included in the same association. If so, that would be worthwhile testing.

And, about run time, this is fairly short, I believe on the order of 20 minutes.

Reply:
I am curious why did you write the RegTestData class rather than just use pytest.fixtures. This regression test seems to be set up differently than the other regression tests. Is it because you wanted to pass the results of the spec2 pipeline to the spec3 pipeline ?

Reply:
Yes, done that way for the chaining. Note however, it is still a `RegtestData` object, just the one returned from `run_spec2` instead of a clean one.  For chaining, this allows isolation and independent testing of the individual pipelines while making it clear that the `run_spec3` fixture is dependent on the result of the `run_spec2` fixture.